src/phpDocumentor/Phpdoc3XmlToArrayConverter.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace phpDocumentor;
+
+final class Phpdoc3XmlToArrayConverter implements XmlConverter
+{
+    /**
+     * @var \SimpleXMLElement
+     */
+    private $xml;
+
+    /**
+     * @param \SimpleXMLElement $xml
+     */
+    public function __construct(\SimpleXMLElement $xml)
```

Comment:
because you pass the xml to the constructor you can't re-use this service (for example: when replaceLocation is used) and you need to create a new service for each xml. This is not necessary as this service does not depend on this state; you can just pass the xml to the convert method and use it there